src/scripts/main.js

```diff
@@ -1,3 +1,36 @@
 'use strict';
 
-// write code here
+const tableNames = document.querySelectorAll('thead th');
+
+tableNames.forEach((element, index) => {
+  element.addEventListener('click', () => sortTable(index));
+});
+
+function sortTable(index) {
+  const tableBody = document.querySelector('tbody');
+  const rows = Array.from(tableBody.querySelectorAll('tr'));
+
+  rows.sort((rowA, rowB) => {
+    const cellA = rowA.cells[index].textContent.trim();
+    const cellB = rowB.cells[index].textContent.trim();
+
+    return compareValues(cellA, cellB, index);
+  });
+
+  tableBody.append(...rows);
+}
+
+function compareValues(cellA, cellB, index) {
+  if (index <= 1) {
+    return cellA.localeCompare(cellB);
+  }
+
+  if (index === 3) {
```

Comment:
The condition `index === 3` assumes that the fourth column contains currency values. Make sure this condition matches the table's actual structure.